ui/src/lib/client.ts

```diff
@@ -5,16 +5,47 @@ import {
 import { redirect } from "next/navigation";
 import { auth } from "../app/api/auth/[...nextauth]/authOptions";
 
-export async function getClient(tenantId: string) {
-  const session = await auth();
+export class LHUserTasksClientSingleton {
+  private static instance: LittleHorseUserTasksApiClient | null = null;
+  private static currentConfig: {
+    baseUrl: string;
+    tenantId: string;
+    accessToken: string;
+  } | null = null;
+
+  private constructor() {}
```

Comment:
usage syntax could be neater by calling `getInstance` inside the constructor and returning the instance, so you can use the singleton with the `new ExampleClass()` syntax.

i suppose it is just preference though

Reply:
Thats a good point you brought up. Can you push those changes to the PR and ill merge.

Reply:
@HazimAr Actually TypeScript class constructors can't return promises so nvm